src/worker.js

```diff
@@ -105,6 +105,14 @@ Module['instantiateWasm'] = (info, receiveInstance) => {
 }
 #endif
 
+// Turn unhandled rejected promises into errors so that the main thread will be
+// notified about them.
+self.onunhandledrejection = (e) => {
+  if (typeof(e.reason) !== 'undefined') {
```

Comment:
When is the reason undefined? (Wouldn't we still want to throw an error then?)

Reply:
Yeah, I think you're right that maybe we should just propagate the error unconditionally and unchanged. @sbc100, did you have ideas about what would be best to do here?

Reply:
Have you ever seen the reason be undefined?   If not, perhaps its reasonable to assert that there is a reason and then throw it?

Reply:
Yes, it seems that if you do `Promise.reject("foo")`, then there is no reason in the event object. From what I've seen, all the system-generated promise rejections have reasons, though.

Reply:
Right... but when if happens we wouldn't want to ignore it would we?  Don't we still want to propagate some kind of error?

Reply:
maybe if there is a `reason` field we should propagate that and otherwise we should propagate the event object directly?

Reply:
That sounds good. Or if there is no reason we could just propagate "rejected" or something generic. The event might be better as it might contain some info I guess.